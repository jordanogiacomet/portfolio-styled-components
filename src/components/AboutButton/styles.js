import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
