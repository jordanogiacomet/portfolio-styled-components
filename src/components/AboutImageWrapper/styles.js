import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
