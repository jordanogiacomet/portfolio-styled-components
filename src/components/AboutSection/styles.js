import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
