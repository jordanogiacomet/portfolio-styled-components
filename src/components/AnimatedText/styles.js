import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;