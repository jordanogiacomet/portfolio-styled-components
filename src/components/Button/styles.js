import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 