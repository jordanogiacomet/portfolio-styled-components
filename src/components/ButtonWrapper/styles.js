import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;