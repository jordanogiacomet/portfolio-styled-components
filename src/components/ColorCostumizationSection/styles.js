// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
