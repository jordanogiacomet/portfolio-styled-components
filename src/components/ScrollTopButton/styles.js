import styled, { css } from 'styled-components';

export const Button = styled.button`
  position: fixed;
  bottom: 10rem;
  left: 4rem;
  width: 40px;
  height: 40px;
  ${({ via, to }) => css`
    background-color: ${via};
    &:hover {
      transition: ease-in-out 0.3s;
      background-color: ${to};
    }
  `}
  color: white;
  border: none;
  border-radius: 50%;
  z-index: 99999999;
  padding: 10px;
  cursor: pointer;
  opacity: ${(props) => (props.isVisible ? '1' : '0')};
  visibility: ${(props) => (props.isVisible ? 'visible' : 'hidden')};
  transition: opacity 0.3s, visibility 0.3s, background-color 0.3s;
  @media (max-width: 768px) {
    display: none;
  }
`;
