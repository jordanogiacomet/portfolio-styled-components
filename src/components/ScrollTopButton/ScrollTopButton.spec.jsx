import { renderTheme } from '../../styles/render-theme';
import ScrollToTopButton from '.';
import { screen, fireEvent } from '@testing-library/react';

describe('<ScrollToTopButton />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<ScrollToTopButton />);
    expect(container).toMatchSnapshot();
  });

  it('should render ScrollToTopButton with icon', () => {
    renderTheme(<ScrollToTopButton />);
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
    expect(button.querySelector('svg')).toBeInTheDocument();
  });

  it('should show button when scrolled down', () => {
    renderTheme(<ScrollToTopButton />);
    fireEvent.scroll(window, { target: { scrollY: 400 } });
    const button = screen.getByRole('button');
    expect(button).toHaveStyle('opacity: 1');
  });

  it('should hide button when scrolled up', () => {
    renderTheme(<ScrollToTopButton />);
    fireEvent.scroll(window, { target: { scrollY: 100 } });
    const button = screen.getByRole('button');
    expect(button).toHaveStyle('opacity: 0');
  });

  it('should scroll to top when clicked', () => {
    renderTheme(<ScrollToTopButton />);
    window.scrollTo = jest.fn();
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(window.scrollTo).toHaveBeenCalledWith({
      top: 0,
      behavior: 'smooth',
    });
  });
});
