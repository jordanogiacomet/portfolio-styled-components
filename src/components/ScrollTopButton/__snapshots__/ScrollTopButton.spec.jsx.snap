// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollTopButton /> should match snapshot 1`] = `
.c0 {
  position: fixed;
  bottom: 10rem;
  left: 4rem;
  width: 40px;
  height: 40px;
  background-color: #4169E1;
  color: white;
  border: none;
  border-radius: 50%;
  z-index: 99999999;
  padding: 10px;
  cursor: pointer;
  opacity: 0;
  visibility: hidden;
  transition: opacity 0.3s,visibility 0.3s,background-color 0.3s;
}

.c0:hover {
  transition: ease-in-out 0.3s;
  background-color: #00008B;
}

@media (max-width: 768px) {
  .c0 {
    display: none;
  }
}

<div>
  <button
    class="c0"
    to="#00008B"
    via="#4169E1"
  >
    <svg
      aria-hidden="true"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.5 10.5 12 3m0 0 7.5 7.5M12 3v18"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
